### DISCUSSION TITLE
Issue with third level imports

### DISCUSSION THREAD

**Discussion by panaotizz** [pyinstaller/pyinstaller#5870] [Help]
Hi everyone,

Im sure this has been posted dozens of times and i searched through the issues but couldn't find a solution.

Im trying to build an executable that uses the 'meraki' module. Installer builds but then when i run the executable i get: 

```
Traceback (most recent call last):
  File "main.py", line 7, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 540, in exec_module
  File "meraki\__init__.py", line 5, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 540, in exec_module
  File "meraki\rest_session.py", line 13, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 540, in exec_module
  File "meraki\__init__.py", line 5, in <module>
ModuleNotFoundError: No module named 'meraki.__init__.rest_session'
```

I tried adding meraki, meraki.__init__ and meraki.__init__.rest_session in the spec file and rebuilding but to no avail.

Any help would be appreciated

**Comment by bwoodsend** [comment]
What is line 7 of `main.py`? I guess it's some kind of `import meraki` but what is it exactly?

**Reply by bwoodsend** [reply]
And are you sure that PyInstaller and `meraki` are in the same environment?

**Reply by rokm** [reply]
A regular `import meraki` seems sufficient to trigger this issue.

It seems that we end up misinterpreting relative imports from [package's `__init__.py`](https://github.com/meraki/dashboard-api-python/blob/master/meraki/__init__.py):

```python
from datetime import datetime
import logging
import os

from .rest_session import *
from .api.organizations import Organizations
from .api.networks import Networks
from .api.devices import Devices
from .api.appliance import Appliance
from .api.camera import Camera
from .api.cellularGateway import CellularGateway
from .api.insight import Insight
from .api.sm import Sm
from .api.switch import Switch
from .api.wireless import Wireless

# Batch class imports
from .api.batch import Batch

[...]
```

So `meraki.rest_session` imported relatively ends up being `meraki.__init__.rest_session`.

**Reply by rokm** [reply]
Hm, actually, I suspect [this](https://github.com/meraki/dashboard-api-python/blob/3956b7c2f677bdc21905980e9b77e8966b94692f/meraki/rest_session.py#L13) is the problematic line that causes `meraki.__init__` to appear among the modules.

Changing that into absolute import (`from meraki import __version__`) and pushing `__version__` above the imports in `__init__.py` seems to fix the problem.

**Reply by panaotizz** [reply]
is there anything i can do on my end or should i wait for a patch ?

Thank you so much for your replies!

**Reply by rokm** [reply]
You can modify your local `meraki` installation as described above.

**Reply by bwoodsend** [reply]
I didn't even think that that line would be legal in Python. Despite appearing to work, it's still not really correct. Looking at `sys.modules`, it has had to load `__init__` twice, once as `meraki` (the normal behaviour) and once a submodule `meraki.__init__` (which should never happen). @panaotizz I suggest you make a pull request to `meraki` to change this.

**Reply by rokm** [reply]
I suspect they ended up with that because it did not cause circular import error (which is why with proper `from meraki import __version__` (or equivalent relative `from . import __version__`), `__version__` in `__init__.py` needs to be pushed above local imports.

**Comment by panaotizz** [comment]
It is import meraki, yes

**Comment by panaotizz** [comment]
Thank you so much again, it worked perfectly. 

I will make a pull request to meraki 

>
